src/hotspot/share/asm/codeBuffer.cpp

```diff
@@ -135,10 +135,11 @@ CodeBuffer::~CodeBuffer() {
     // Previous incarnations of this buffer are held live, so that internal
     // addresses constructed before expansions will not be confused.
     cb->free_blob();
+  }
+  if (_overflow_arena != nullptr) {
     // free any overflow storage
-    delete cb->_overflow_arena;
```

Comment:
With this change, only this->_overflow_arena will be non-null after exapdn().  In expand() we won't copy this value to the buffer in _before_expand, so we can move this delete outside the loop like before.

Reply:
I moved it outside. I only added `!= nullptr` check which may look like we still inside loop.

Reply:
Sorry, I read it wrong.